libraries/AP_Scripting/generator/src/main.c

```diff
@@ -1445,8 +1504,14 @@ void emit_userdata_method(const struct userdata *data, const struct method *meth
     arg = arg->next;
   }
 
+  const char *ud_name = (data->flags & UD_FLAG_LITERAL)?data->name:"ud";
+
   if (data->flags & UD_FLAG_SEMAPHORE) {
-    fprintf(source, "    ud->get_semaphore().take_blocking();\n");
+    fprintf(source, "    %s->get_semaphore().take_blocking();\n", ud_name);
+  }
+
+  if (data->flags & UD_FLAG_SEMAPHORE_POINTER) {
+    fprintf(source, "    %s->get_semaphore()->take_blocking();\n", ud_name);
```

Comment:
```suggestion
    fprintf(source, "    %s->get_semaphore().take_blocking();\n", ud_name);
  } else if (data->flags & UD_FLAG_SEMAPHORE_POINTER) {
    fprintf(source, "    %s->get_semaphore()->take_blocking();\n", ud_name);
```

Let's make it more obvious that these cases are mutual exclusive.

Reply:
fixed